lib/agent.js

```diff
@@ -68,7 +67,11 @@ Agent.prototype._config = function (opts) {
     ? parseUrl(this._conf.serverUrl).hostname
     : 'localhost'
 
-  logger.init({level: this._conf.logLevel})
+  this.logger = this._conf.logger || ConsoleLogLevel
+
+  if (typeof this.logger === 'function') {
+    this.logger = this.logger({ level: this._conf.logLevel })
```

Comment:
Is it safe to assume that all loggers that are a function will have this API?

Reply:
I meant for this to just be a possible initialization option. If you have a logger that doesn't initialize this way, you can just initialize it externally and provide the constructed object instead.

Reply:
I'm thinking if a simpler API simply would be to require people to initialize their loggers before hand?